internals/cli/cli.go

```diff
@@ -58,8 +58,9 @@ var ErrExtraArgs = fmt.Errorf("too many arguments for command")
 
 // CmdOptions exposes state made accessible during command execution.
 type CmdOptions struct {
-	Client *client.Client
-	Parser *flags.Parser
+	Client     *client.Client
+	Parser     *flags.Parser
+	RunOptions *RunOptions
```

Comment:
Since this struct is already an options container for commands, could we not simply pass in the options we need, in a way that is without consumer context ? I believe the only options we need to add for commands are:

```socketPath```
```pebbleDir```

Why do we not simply add these?

Reply:
Yeah, that's a good point, Fred -- it's simpler to reason about and avoids mixing concerns. Let's do that.

Reply:
With the new `ParserOptions` struct and embedding `*ParserOptions` into `CmdOptions`, this practically does the same thing (replaces the existing `Client` with the same name + adds `SocketPath` and `PebbleDir`).

However, we also need `ParserOptions` in `cmd_enter.go`'s `cmdEnter` (because we create a new parser there), so I opted for keeping `*ParserOptions` in `CmdOptions` as embedded struct, so there I can just say: `Parser(cmd.parserOptions)`.

If that is not to your liking, I can also explicity add the fields as in:

```go
type CmdOptions struct {
        Client     *client.Client
        Parser     *flags.Parser
        PebbleDir  string
        SocketPath string
}
```

It would be relatively easy to change `&CmdOptions{Parser: parser, ParserOptions: opts}` to explicitly grab client, pebble dir and socket path from `ParserOptions` and put them into `CmdOptions`.

And on the `cmd_enter.go` side it wouldn't be too much trouble to add all 3 fields (`client`, `pebbleDir`, `socketPath`) to `cmdEnter` (instead of just `parserOptions *ParserOptions`) and then when creating the parser, doing it explicitly:

```go
        parser := Parser(&ParserOptions{
                Client: cmd.client,
                SocketPath: cmd.socketPath,
                PebbleDir: cmd.pebbleDir,
        })
```

Slightly more maintenance work when adding or removing fields from `ParserOptions` (and it's easier to forget wiring all values through `ParserOptions` -> `Parser()` -> `CmdOptions` -> `cmdEnter` -> `ParserOptions` in `cmd_enter.go`?) but potentially more explicit and separated.

Opinions?

Reply:
Thanks. Yeah, I don't like the ParserOptions stuffed into CmdOptions -- I think we should spell them out like you have in the struct definition in your comment above. Tiny bit more wiring, but conceptually cleaner. Same with cmd_enter -- let's spell out those fields like you have with cmd_run.